tests/modules/programs/zsh/local-variables.nix

```diff
@@ -0,0 +1,40 @@
+{ lib, ... }:
+
+with lib;
+
+{
+  config = {
+    programs.zsh = {
+      enable = true;
+
+      localVariables = rec {
+        V1 = true;
+        V2 = false;
+        V3 = "some-string";
+        V4 = 42;
+        V5 = builtins.attrValues V6;
+        V6 = {
+          a = V1;
+          b = V2;
+          c = V3;
+          d = V4;
+        };
+      };
+    };
+
+    test.stubs.zsh = { };
+
+    nmt.script = ''
+      assertFileExists home-files/.zshrc
+      assertFileContains home-files/.zshrc 'V1=true'
```

Comment:
Will this pass if `typeset -A V1=true` is in the file?

Reply:
Ah, yes it probably will. I'll switch these to something like `^V1=true$`.

But I won't be able to do that for `V5` and `V6` because those contain `(` and `[` characters that I may or may not need to escape depending on which regex dialect is used by grep. And I'm unable to control that as that's handled by nmt...

Reply:
Hmm, looks I could address this for `V5` at least by wrapping the braces in square brackets as an escaping mechanism independent of dialect (`[(]` and `[)]`) (works in POSIX basic, POSIX extended and Perl compatible regular expressions apparently).